ros_buildfarm/config/release_build_file.py

```diff
@@ -43,6 +43,12 @@ def __init__(self, name, data):
             self.abi_incompatibility_assumed = \
                 bool(data['abi_incompatibility_assumed'])
 
+        self.build_environment_variables = {}
```

Comment:
Initializing this to an empty dict made sense initially, but it creates some similar but disparate constructs later as this is ultimately passed through via argparse nargs, which initializes omitted arguments to None rather than an empty list. So we end up with two lines

https://github.com/ros-infrastructure/ros_buildfarm/pull/554/files#diff-041d2fd48de98faa6a1db2d571bea72cR103 and https://github.com/ros-infrastructure/ros_buildfarm/pull/554/files#diff-b7ddb841a26f20ca5e4325409de758b3R107 which seem like they should be the same but are slightly different (one checks for presence / absence and the other for a non-zero length.

Initializing to None would allow us to keep that consistent but I'm not sure what is preferred.

Reply:
The other options in this file initialize with `None` and only set the value to something else when the config file contains the specific key. So I would use the same here.

Both conditions can be written in the same way since an empty dictionary as well as `None` are consider `False` when interpreted as a `bool`.